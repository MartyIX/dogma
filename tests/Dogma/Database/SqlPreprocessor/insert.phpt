<?php

/**
 * Test: Nette\Database\SqlPreprocessor: insert
 *
 * @author     Vlasta Neubauer
 * @package    Nette\Database
 * @subpackage UnitTests
 */

require_once __DIR__ . '/../connect.inc.php';


$processor = $connection->getSqlPreprocessor();


Assert::same(array("INSERT INTO table (col1, col2, col3) VALUES (123, 'abc', '2012-01-16 00:00:00')", array()), 
	$processor->process("INSERT INTO table (col1, col2, col3) VALUES (?)", 
		array(
			array(123, 'abc', new DateTime('2012-01-16'))
		)
	)
);

Assert::same(array("INSERT INTO table (col1, col2, col3) VALUES (123, 'abc', '2012-01-16 00:00:00')", array()), 
	$processor->process("INSERT INTO table (col1, col2, col3) VALUES ?", 
		array(
			array(123, 'abc', new DateTime('2012-01-16'))
		)
	)
);

Assert::same(array("INSERT INTO table (`col1`, `col2`, `col3`) VALUES (123, 'abc', '2012-01-16 00:00:00')", array()), 
	$processor->process("INSERT INTO table ?", 
		array(
			array('col1' => 123, 'col2' => 'abc', 'col3' => new DateTime('2012-01-16'))
		)
	)
);

Assert::same(array("INSERT INTO table (col1, col2, col3) VALUES (123, 'abc', '2012-01-16 00:00:00')", array()), 
	$processor->process("INSERT INTO table (col1, col2, col3) VALUES ", 
		array(
			array(123, 'abc', new DateTime('2012-01-16'))
		)
	)
);

Assert::same(array("INSERT INTO table (`col1`, `col2`, `col3`) VALUES (123, 'abc', '2012-01-16 00:00:00')", array()), 
	$processor->process("INSERT INTO table ", 
		array(
			array('col1' => 123, 'col2' => 'abc', 'col3' => new DateTime('2012-01-16'))
		)
	)
);
