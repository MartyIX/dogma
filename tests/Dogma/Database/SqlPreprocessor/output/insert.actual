array(
	'INSERT INTO table (col1, col2, col3) VALUES ((`0`, `1`, `2`) VALUES (123, \'abc\', \'2012-01-16 00:00:00\'))',
	array(),
)