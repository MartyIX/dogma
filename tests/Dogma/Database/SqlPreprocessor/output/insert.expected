array(
	'INSERT INTO table (col1, col2, col3) VALUES (123, \'abc\', \'2012-01-16 00:00:00\')',
	array(),
)